teamworkApp/lib/stackedDistribution.py

```diff
@@ -0,0 +1,111 @@
+# muddersOnRails()
+# Maeve Murphy November 15, 2017
+# Last updated: 11-15-17
+
+# creates a histogram that shows the number of students with
+# primary, secondary, tertiary, and quarternary rankings
+# of each type.
+
+# resources:
+# https://matplotlib.org/examples/pylab_examples/bar_stacked.html
+
+import matplotlib.pyplot as plt
+import numpy as np
+from enum import IntEnum
+import sqlite3
+
+DB = 'db/development.sqlite3'
+Style = IntEnum('Style', 'Communicator, Collaborator, Challenger, Contributor', start=0)
+
+def get_style_tuples():
+    """Retrieve all the style data stored in the database"""
+    conn = sqlite3.connect(DB)
+    c = conn.cursor()
+    c.execute(
+        'SELECT communicator, collaborator, challenger, contributor FROM styles'
+    )
+    style_data = c.fetchall()
+    conn.commit()
+    conn.close()
+
+    return style_data
+
+def tally_nth(style_data, n):
+    """given a tuple of 4 scores: (communicator, collaborator,
+    challenger, contributor), return tallies for how many
+    students in the DB have each as their nth order prominence of style.
+    For example, let n = 1 and get back the highest order prominence (primary)
+        style
+    NOTE: ties are settled such that they are counted in both categories of the
+    tie. For example, if the student's style was [45, 43, 43, 49] collaborator
+    would be 2nd and 3rd and challenger would *also* be 2nd and 3rd"""
+    # communicator, collaborator, challenger, contributor
+    tallies = [0,0,0,0]
+    for student in style_data:
+        # sort from low to high
+        sorted_styles = sorted(student)
+        # max is -1, second greatest is -2, etc.
+        if student[0] == sorted_styles[-n]:
+            tallies[0] += 1
+        if student[1] == sorted_styles[-n]:
+            tallies[1] += 1
+        if student[2] == sorted_styles[-n]:
+            tallies[2] += 1
+        if student[3] == sorted_styles[-n]:
+            tallies[3] += 1
+    return tallies
```

Comment:
I think that this would be cleaner with some list comprehensions since right now you have a lot of duplicate code. Something like:
```python
indexes = [i for i, x in enumerate(student) if x == sorted_styles[-n]]
for i in index:
    tallies[i] += 1
```

Reply:
Done!